packages/xo-web/src/common/shallow-equal.js

```diff
@@ -27,6 +27,20 @@ const shallowEqual = (c1, c2) => {
     return true
   }
 
+  if (type === 'set') {
+    if (c1.size !== c2.size) {
+      return false
+    }
+
+    for (const v of c1) {
+      if (c2.has(v)) {
```

Comment:
I think it's the inverse. 

`c2` isn't equal to `c1` if it's not contain one of its value.